rmw_opensplice_cpp/src/functions.cpp

```diff
@@ -508,34 +591,89 @@ rmw_create_client(const rmw_node_t * node,
                   const rosidl_service_type_support_t * type_support,
                   const char * service_name)
 {
-  return NULL;
+  if (node->implementation_identifier != opensplice_cpp_identifier)
+  {
+    rmw_set_error_string("node handle not from this implementation");
+    // printf("but from: %s\n", node->implementation_identifier);
+    return NULL;
+  }
+
+  DDS::DomainParticipant_ptr participant = \
+    static_cast<DDS::DomainParticipant_ptr>(node->data);
+
+  const service_type_support_callbacks_t * callbacks = \
+    static_cast<const service_type_support_callbacks_t *>(type_support->data);
+
+  DDS::DataReader_ptr response_datareader;
+
+  void * requester = callbacks->create_requester(
+    participant, service_name, reinterpret_cast<void **>(&response_datareader));
+
+  OpenSpliceStaticClientInfo* client_info = new OpenSpliceStaticClientInfo();
+  client_info->requester_ = requester;
+  client_info->callbacks_ = callbacks;
+  client_info->response_datareader_ = response_datareader;
+
+  rmw_client_t * client = rmw_client_allocate();
+  client->implementation_identifier = opensplice_cpp_identifier;
+  client->data = client_info;
+  return client;
 }
 
 rmw_ret_t
 rmw_destroy_client(rmw_client_t * client)
 {
-  return RMW_RET_OK;
+  if (client) {
+    // TODO(esteve): de-allocate Requester and response DataReader
```

Comment:
It's not done, I'll add a new function to the callbacks structure to destroy the Requester in a typesafe way.